ansys/dpf/core/animator.py

```diff
@@ -80,39 +81,51 @@ def render_field(frame):
                            **kwargs)
             kwargs_in = _sort_supported_kwargs(
                 bound_method=self._plotter.add_text, **freq_kwargs)
-            str_template = "t={0:{2}} {1}"
             self._plotter.add_text(str_template.format(loop_over.data[frame], unit, freq_fmt),
                                    **kwargs_in)
             if cpos:
                 self._plotter.camera_position = cpos[frame]
 
         try:
-            # Write initial frame
-            render_field(0)
-            # If not off_screen, enable the user to choose the camera position
-            if not kwargs.pop("off_screen", None):
-                print('Orient the view, then press "q" to close the window '
-                      'and produce an animation')
-            # Show is necessary even when off_screen to initiate the renderer
-            result = self.show_figure(auto_close=False, **kwargs)
-            if save_as:
-                try:
-                    self._plotter.write_frame()
-                except AttributeError as e:  # pragma: no cover
-                    if "To retrieve an image after the render window has been closed" in e.args[0]:
-                        print("Animation canceled.")
-                        return result
-            # For each additional frame requested
-            if len(indices) > 1:
-                for frame in range(1, len(indices)):
+            def animation():
+                if save_as:
                     try:
-                        render_field(frame)
+                        self._plotter.write_frame()
                     except AttributeError as e:  # pragma: no cover
-                        if "'NoneType' object has no attribute 'interactor'" in e.args[0]:
+                        if "To retrieve an image after the render window has been closed" \
+                                in e.args[0]:
                             print("Animation canceled.")
+                            print(e)
                             return result
-                    if save_as:
-                        self._plotter.write_frame()
+                # For each additional frame requested
+                if len(indices) > 1:
+                    for frame in range(1, len(indices)):
+                        try:
+                            render_frame(frame)
+                        except AttributeError as e:  # pragma: no cover
+                            if "'NoneType' object has no attribute 'interactor'" in e.args[0]:
+                                print("Animation canceled.")
+                                return result
+                        if save_as:
+                            self._plotter.write_frame()
+
+            # Write initial frame
+            render_frame(0)
+            # If not off_screen, enable the user to choose the camera position
+            off_screen = kwargs.pop("off_screen", None)
+            if off_screen is None:
```

Comment:
@PProfizi that means that we set off_screen by defualt. Is that what we want ?

Reply:
@anslpa I do not see how this sets `off_screen=True` by default. This takes the value in `kwargs`, if not present then it is set to `None`, and if `None `then we do a last check of the `pv.OFF_SCREEN` variable which might have been set globally to `True`, for example when running tests or examples in our pipelines.
I tested it and to me it is `False `by default.